blockchain/state/statedb.go

```diff
@@ -225,6 +225,16 @@ func (self *StateDB) AddRefund(gas uint64) {
 	self.refund += gas
 }
 
+// SubRefund removes gas from the refund counter.
+// This method will panic if the refund counter goes below zero
+func (self *StateDB) SubRefund(gas uint64) {
+	self.journal.append(refundChange{prev: self.refund})
+	if gas > self.refund {
```

Comment:
I have a question.
Is it the right order to check if the refund counter is below zero after the refundChange append?

Reply:
@hyochan-brown um.. I'm not sure i understood your question, however, refund counter is subtracted after check is done (line 235)

Reply:
I was wondering if 231 line are okay to be higher than 232...

Reply:
Oh.. At least i know, self.journal.append marks dirty flag, also it panics when gas > self.refund.